### PR TITLE
[BUG] Expected results rework and Shapelet Transform bugfix

### LINKED ISSUE #711: [BUG] Intermittent test fail from shapelet transform
### Describe the bug

the test, 
aeon/transformations/collection/tests/test_shapelet_transform.py::test_st_on_basic_motions

fails occasionally with results different from expected. Possible obscure lack of seeding, perhaps when shapelets are equal? Both of below are on ubuntu 22.04, but different python versions. If we see more, post link here

### Steps/Code to reproduce the bug

see
https://github.com/aeon-toolkit/aeon/actions/runs/6040287686/job/16391060741?pr=673
and
https://github.com/aeon-toolkit/aeon/actions/runs/6019651707/job/16329767004

### Expected results

not fail tests. 

### Actual results

fails test

### Versions

_No response_

### PR DESCRIPTION
**Pull request by MatthewMiddlehurst**
Fixes #711.

Multiple changes in this PR made during the search for the ST bug.
- Moves the RandomState object generation for multiple estimators which use joblib outside of the threaded function. This reduces the amount of lines used for each and is cleaner IMO.
- Reworks the methods for generating expected results and the tests which use them (as a result of the above). The UnitTest check now uses less data, and the different tests for different datasets have been merged into one.
- Adds an expected results test in line for the classification and regression ones.
- Regenerates the expected results where relevant.
- Rounds information gain to 8 decimals in the shapelet transform (the actual bugfix) and changes shapelet sorting so shorter shapelets are kept during information gain ties.

This PR was initially hunting for the bugfix, and regenerating expected results while doing that was very annoying so i made it easier 🙂.

After 10 reruns, the ST bug did not reappear (other failures happened, but not related to this).

I can split this PR up on request, since some of the changes are not really related.

### CODE REVIEW

**Comment by TonyBagnall** [comment]
is this not now completely unseeded though?

**Comment by MatthewMiddlehurst** [comment]
Instead of creating the `RandomState` inside the joblib thread/function, this creates it outside and passes it as a parameter. 

The number generated by rng in `check_random_state(rng.randint(np.iinfo(np.int32).max))` to create the new seeded `RandomState` should always be the same if rng is seeded, or random if it is not.

...Of course this changes the rng previously used, so all the old results have to be updated 😢

**Comment by MatthewMiddlehurst** [comment]
also this could just not be the issue and its still broken, but this method may be better either way

**Comment by TonyBagnall** [comment]
> also this could just not be the issue and its still broken, but this method may be better either way

I think it is better, I think the other way was to avoid seeding the shapelet in the same way, but it seems a bit mad to create a new one for every shapelet.

File: aeon/transformations/collection/shapelet_based/_shapelet_transform.py [hunk 1]
```diff
@@ -325,7 +329,7 @@ def _fit(self, X, y=None):
             for s in class_shapelets
             if s[0] > 0
         ]
-        self.shapelets.sort(reverse=True, key=lambda s: (s[0], s[1], s[2], s[3], s[4]))
+        self.shapelets.sort(reverse=True, key=lambda s: (s[0], -s[1], s[2], s[3], s[4]))
```
Comment (TonyBagnall) [inline aeon/transformations/collection/shapelet_based/_shapelet_transform.py:332]: this changes ST to favour short shapelets when IG is tied, which is sensible (adding to remind me!)

File: aeon/transformations/collection/shapelet_based/_shapelet_transform.py [hunk 2]
```diff
@@ -437,7 +438,7 @@ def _extract_random_shapelet(self, X, y, i, shapelets, max_shapelets_per_class):
             worst_quality,
         )
 
-        return quality, length, position, dim, inst_idx, cls_idx
+        return np.round(quality, 8), length, position, dim, inst_idx, cls_idx
```
Comment (TonyBagnall) [inline aeon/transformations/collection/shapelet_based/_shapelet_transform.py:441]: this may well have been cause of the bug

**Review by TonyBagnall — APPROVED** [review]

**Comment by TonyBagnall** [comment]
one more repeat for good luck

**Pull request merged**